packages/flutter_tools/lib/src/base/dds.dart

```diff
@@ -4,220 +4,42 @@
-Future<DartDevelopmentServiceInstance> defaultStartDartDevelopmentService(
-  Uri remoteVmServiceUri, {
-  required bool enableAuthCodes,
-  required bool ipv6,
-  required bool enableDevTools,
-  required List<String> cachedUserTags,
-  Uri? serviceUri,
-  String? google3WorkspaceRoot,
-  Uri? devToolsServerAddress,
-}) async {
-  final String exe = globals.artifacts!.getArtifactPath(
-    Artifact.engineDartBinary,
-  );
-  final io.Process process = await io.Process.start(
-    exe,
-    <String>[
-      'development-service',
-      '--vm-service-uri=$remoteVmServiceUri',
-      if (serviceUri != null) ...<String>[
-        '--bind-address=${serviceUri.host}',
-        '--bind-port=${serviceUri.port}',
-      ],
-      if (!enableAuthCodes) '--disable-service-auth-codes',
-      if (google3WorkspaceRoot != null)
-        '--google3-workspace-root=$google3WorkspaceRoot',
-      for (final String tag in cachedUserTags) '--cached-user-tags=$tag',
-    ],
-  );
-  final Completer<DartDevelopmentServiceInstance> completer =
-      Completer<DartDevelopmentServiceInstance>();
-  late StreamSubscription<Object?> stderrSub;
-  stderrSub = process.stderr
-      .transform(utf8.decoder)
-      .transform(json.decoder)
-      .listen((Object? result) {
-    if (result
-        case {
-          'state': 'started',
-          'ddsUri': final String ddsUriStr,
-        }) {
-      final Uri ddsUri = Uri.parse(ddsUriStr);
-      final String? devToolsUriStr = result['devToolsUri'] as String?;
-      final Uri? devToolsUri =
-          devToolsUriStr == null ? null : Uri.parse(devToolsUriStr);
-      final String? dtdUriStr =
-          (result['dtd'] as Map<String, Object?>?)?['uri'] as String?;
-      final Uri? dtdUri = dtdUriStr == null ? null : Uri.parse(dtdUriStr);
+export 'package:dds/dds.dart' show DartDevelopmentServiceException;
 
-      completer.complete((
-        process: process,
-        serviceUri: ddsUri,
-        devToolsUri: devToolsUri,
-        dtdUri: dtdUri,
-      ));
-    } else if (result
-        case {
-          'state': 'error',
-          'error': final String error,
-        }) {
-      final Map<String, Object?>? exceptionDetails =
-          result['ddsExceptionDetails'] as Map<String, Object?>?;
-      completer.completeError(
-        exceptionDetails != null
-            ? DartDevelopmentServiceException.fromJson(exceptionDetails)
-            : StateError(error),
-      );
-    } else {
-      throw StateError('Unexpected result from DDS: $result');
-    }
-    stderrSub.cancel();
-  });
-  return completer.future;
-}
-
-typedef DDSLauncherCallback = Future<DartDevelopmentServiceInstance> Function(
-  Uri remoteVmServiceUri, {
-  required bool enableAuthCodes,
-  required bool ipv6,
-  required bool enableDevTools,
-  required List<String> cachedUserTags,
-  Uri? serviceUri,
-  String? google3WorkspaceRoot,
-  Uri? devToolsServerAddress,
+typedef DDSLauncherCallback = Future<DartDevelopmentServiceLauncher> Function({
```

Comment:
As a TODO, should this typedef be moved into package:dds? Since I'm thinking this would need to match `DartDevelopmentServiceLauncher.start` which lives there, am I right?

Reply:
ahh yeah, i guess if it lived in dds but you changed it, it would still be a breaking change here since we implement it. nvm...